[sqlancer] create view failed

<!-- (At least include the following, feel free to add if you have more content) -->

### Steps to reproduce the behavior (Required)
```
DROP DATABASE IF EXISTS v19;
CREATE DATABASE v19;
USE v19;
admin set frontend config ("enable_decimal_v3" = "true");
CREATE TABLE IF NOT EXISTS t0 (c_0_0 DATETIME NOT NULL ,c_0_1 VARCHAR(31) NULL ,c_0_2 BOOLEAN NULL ,c_0_3 DECIMAL(16, 12)  NOT NULL ,c_0_4 FLOAT SUM NOT NULL ,c_0_5 DATETIME MIN NOT NULL ,c_0_6 PERCENTILE PERCENTILE_UNION NOT NULL ,c_0_7 HLL HLL_UNION NULL ,c_0_8 PERCENTILE PERCENTILE_UNION NOT NULL ,c_0_9 LARGEINT MIN NULL ,c_0_10 SMALLINT MAX NULL , INDEX index1_c_0_2(`c_0_2`) USING BITMAP) AGGREGATE KEY (c_0_0,c_0_1,c_0_2,c_0_3) PARTITION BY RANGE(c_0_0) (START ("2010-01-01") END ("2021-12-31") EVERY (INTERVAL 121 day)) DISTRIBUTED BY HASH (c_0_2,c_0_1) BUCKETS 3 properties("replication_num"="3") ;
CREATE TABLE t1 (c_1_0 CHAR(1) NULL ,c_1_1 VARCHAR(41) NULL ,c_1_2 DATE NULL ,c_1_3 DATETIME NULL ,c_1_4 ARRAY<INT> NULL ,c_1_5 DATE NULL ,c_1_6 INT NOT NULL ,c_1_7 DECIMAL(6, 4)  NOT NULL ,c_1_8 BOOLEAN NULL ,c_1_9 ARRAY<INT> NULL , INDEX index1_c_1_0(`c_1_0`) USING BITMAP) UNIQUE KEY (c_1_0,c_1_1,c_1_2,c_1_3) PARTITION BY RANGE(c_1_3) (START ("2010-01-01") END ("2021-12-31") EVERY (INTERVAL 151 day)) DISTRIBUTED BY HASH (c_1_2,c_1_3,c_1_1) BUCKETS 3 properties("replication_num"="3") ;
CREATE TABLE t2 (c_2_0 DATETIME NOT NULL ,c_2_1 CHAR(31) NULL ,c_2_2 ARRAY<INT> NOT NULL ,c_2_3 DATE NOT NULL ) DUPLICATE KEY (c_2_0,c_2_1) PARTITION BY RANGE(c_2_0) (START ("2010-01-01") END ("2021-12-31") EVERY (INTERVAL 106 day)) DISTRIBUTED BY HASH (c_2_0,c_2_1) BUCKETS 3 properties("replication_num"="3", "bloom_filter_columns"="c_2_1") ;
CREATE TABLE IF NOT EXISTS t3 (c_3_0 DATETIME NOT NULL,c_3_1 DATETIME NOT NULL,c_3_2 BOOLEAN NOT NULL,c_3_3 VARCHAR(21) NOT NULL,c_3_4 DATETIME NOT NULL,c_3_5 DATE NOT NULL,c_3_6 DATETIME NOT NULL,c_3_7 ARRAY<INT> NOT NULL,c_3_8 ARRAY<INT> NOT NULL,c_3_9 FLOAT NOT NULL, INDEX index1_c_3_4(`c_3_4`) USING BITMAP) PRIMARY KEY (c_3_0,c_3_1,c_3_2,c_3_3,c_3_4) PARTITION BY RANGE(c_3_0) (START ("2010-01-01") END ("2021-12-31") EVERY (INTERVAL 104 day)) DISTRIBUTED BY HASH (c_3_2,c_3_4) BUCKETS 3 properties("replication_num"="3") ;
INSERT INTO t1 VALUES ('', '', '2010-01-21', '2010-01-02 08:00:14', [75486280, 1642398784, 1, 418580368, 1720529812, 1733100614, 1733100614, 214774895], '2010-01-06', 26, 0.0000, false, [-2080229560, 1378457264, -145601074, 75486280, -145601074, -689397735, 1241629178, 1378457264]);
INSERT INTO t3 VALUES ('2010-01-20 08:00:14', '2010-01-23 08:00:14', true, '4xBUp7M', '2010-01-23 08:00:14', '2010-01-02', '2009-12-31 08:00:14', [1718274261, -1998242557, 402006228, 1844601272, -118141023, 75486280, -1801635113], [-118141023, -1082942227, 1718274261, 1235798379, -1377008292, 1441071915, -1801635113], 0.4775324221420195);
INSERT INTO t1 VALUES (NULL, 'FCQDJODA8', '2010-01-06', '2010-01-11 08:00:14', [1263312921, -265368778, 1202824436, 601260747, -1377008292, 2143244553, -1559687830, 1986959824, -1312223001, 1642398784, -990671231], '2010-01-01', 0, 0.0000, false, [1508926990, -90039011, 994825458, -145601074, 1844601272, 419042481, -2099187603, 994825458, -145601074, 1506105844, 1378457264]);
TRUNCATE TABLE t0;
INSERT INTO t3 VALUES ('2010-01-25 08:00:14', '2010-01-10 08:00:14', true, 'e', '2010-01-07 08:00:14', '2009-12-31', '2010-01-10 08:00:14', [515089530, 1642398784, -1469970147, 1967863530], [1967863530, -1518676971, 1642398784, 1378457264, -1468717528, -2047824166], 0.24090150989306902);
INSERT INTO t0 VALUES ('2010-01-17 08:00:14', 'Ft0', false, 6.82E-10, 1.50892699E9, '2010-01-03 08:00:14', percentile_hash('839915868'), hll_hash('L6D98l'), percentile_hash('-2123081391'), -730831738, 6);
INSERT INTO t3 VALUES ('2010-01-07 08:00:14', '2010-01-21 08:00:14', true, '4gtwU', '2010-01-22 08:00:14', '2010-01-13', '2010-01-16 08:00:14', [-1414063668, -1522614302, -1802998858, 1681144300, 1275915775, -508297574, 923898662], [1844601272, 1720529812, -1746305626, 1642398784, 994385629], -1.802998858E9), ('2010-01-21 08:00:14', '2010-01-02 08:00:14', true, '', '2010-01-28 08:00:14', '2010-01-28', '2010-01-27 08:00:14', [], [-1518676971, 839915868], 0.26812071476302357);
INSERT INTO t1 VALUES ('', 'd', NULL, '2010-01-16 08:00:14', [261067171, 418580368, 402006228, -106202374, -421224718, -90039011, 1367105786, 994385629], '2010-01-04', 14, 0.0031, true, [2117688323, 994825458, 1792219215, -1294164484, -19899934, -1349697796, 374610337, -640638236]);
INSERT INTO t1 VALUES ('', 'LyE5bE', '2010-01-31', NULL, [468515708, 1681837426, -1559687830, -169654725, 2038162478, 1651615261, 1415083796, 1159733985, -544614646, 1036009216, -992432237, 1967863530, 446657964, 234924751, -2102751603], '2010-01-27', 23, 0.0004, false, [-1743707285, -988783705, -2135357453, -1522614302, 364952637, 1202824436]), ('', '', '2010-01-22', '2010-01-29 08:00:14', [-14674771, -146662779, -1518676971, 1354451368, -1349697796, -897556246, 492722675, 364952637, -1709961858], '2010-01-28', 3, 0.0098, NULL, [-145601074]);
INSERT INTO t3 VALUES ('2010-01-12 08:00:14', '2010-01-09 08:00:14', true, 'gXPj', '2010-01-05 08:00:14', '2010-01-29', '2009-12-31 08:00:14', [990458641, 1755607041, 468515708, -1414063668, -2048150725, -858543038, -1294164484, 2009755197, -1746305626], [261067171, -764213410, 1967863530, -569516214, -1545001761, 1510261420], 0.7973744164419383), ('2010-01-24 08:00:14', '2010-01-31 08:00:14', false, '4', '2010-01-18 08:00:14', '2010-01-08', '2010-01-05 08:00:14', [-1522614302, -1802998858, 418580368], [1378457264, -1619982004, 1681144300, -1619982004, 2092456334, 1488644631, 1625843726, 2127936354, 529274631, -1175265817, -843160811, -1618085560], 0.44761691969393336);
INSERT INTO t0 VALUES ('2010-01-28 08:00:14', '', true, 5.334E-9, -1.294164484E9, '2010-01-11 08:00:14', percentile_hash('1681144300'), hll_hash(''), percentile_hash('-1955028314'), -1709961858, 3), ('2010-01-13 08:00:14', 'DWtJT6eI', true, 9.731E-9, 0.701450971302718, '2010-01-26 08:00:14', percentile_hash('-442160369'), hll_hash('WZWaza'), percentile_hash('-508128002'), -1684998214, NULL);
INSERT INTO t1 VALUES ('', '', '2010-01-02', '2010-01-19 08:00:14', [-1290567354, 1651615261, -1226489857, 2029167030, -988783705, 468515708, 1441071915], '2010-01-29', 16, 0.0021, false, []), ('', '', '2010-01-07', NULL, [-34045874, 1136310612, -1522614302], '2010-01-20', 2, 0.0003, true, [1672187148, 1100061363, 391682383]), ('', NULL, '2010-01-12', NULL, [419042481, -806424253, -1258289518, -544614646, 1508926990, 1275915775, -1469970147], '2010-01-03', 31, 0.0023, true, [-987436687, 2060753351, -317130466, 2117688323, -1743707285, 1263312921, -287895483, 2082763540, 1834363370, -1559687830]);
CREATE VIEW v0 AS WITH with_t_0 as (SELECT t3_3358.c_3_0, t3_3358.c_3_1, t3_3358.c_3_2, t3_3358.c_3_3, t3_3358.c_3_4, t3_3358.c_3_5, t3_3358.c_3_6, t3_3358.c_3_9 FROM t3 AS t3_3358) , with_t_1 as (SELECT t3_3358.c_3_0, t3_3358.c_3_1, t3_3358.c_3_2, t3_3358.c_3_3, t3_3358.c_3_4, t3_3358.c_3_5, t3_3358.c_3_6, t3_3358.c_3_9 FROM t3 AS t3_3358 WHERE (t3_3358.c_3_9) IN ((SELECT t1_3356.c_1_0 FROM t3 AS t3_3358, t2 AS t2_3357,  t1 AS t1_3356 LEFT SEMI JOIN (SELECT t0_3355.c_0_0, t0_3355.c_0_1, t0_3355.c_0_2, t0_3355.c_0_3, t0_3355.c_0_4, t0_3355.c_0_5, t0_3355.c_0_9, t0_3355.c_0_10 FROM t0 AS t0_3355 WHERE false) subt0 ON t1_3356.c_1_2 = subt0.c_0_1 AND '1969-12-17' != subt0.c_0_1 WHERE (('1970-01-05') IN ('1970-01-03') ) IN (true) ) ) ) SELECT subt3.c_3_0 FROM t0 AS t0_3355, t1 AS t1_3356, with_t_0 AS with_t_0, with_t_1 AS with_t_1,  (SELECT t3_3358.c_3_0, t3_3358.c_3_1, t3_3358.c_3_2, t3_3358.c_3_3, t3_3358.c_3_4, t3_3358.c_3_5, t3_3358.c_3_6, t3_3358.c_3_9 FROM t3 AS t3_3358 ORDER BY (((('')||(CASE  WHEN (t3_3358.c_3_6)  BETWEEN (t3_3358.c_3_6) AND (t3_3358.c_3_6)  THEN ''  WHEN false THEN (('')||('')) END))) IS NULL)) subt3 LEFT ANTI JOIN (SELECT t2_3357.c_2_0, t2_3357.c_2_1, t2_3357.c_2_3 FROM t2 AS t2_3357) subt2 ON subt3.c_3_2 = subt2.c_2_1 AND subt3.c_3_2 > (('5')||('')) AND subt3.c_3_9 < subt2.c_2_0 AND subt3.c_3_9 <= subt2.c_2_0;

```

```
DROP DATABASE IF EXISTS v2;
CREATE DATABASE v2;
USE v2;
admin set frontend config ("enable_decimal_v3" = "true");
CREATE TABLE IF NOT EXISTS t0 (c_0_0 BOOLEAN NOT NULL ,c_0_1 DATE NULL ,c_0_2 DECIMAL(29, 6)  NULL ,c_0_3 BOOLEAN NOT NULL ,c_0_4 BOOLEAN NOT NULL ,c_0_5 VARCHAR(21) NOT NULL ,c_0_6 TINYINT NULL ,c_0_7 ARRAY<STRING> NULL ,c_0_8 DATETIME NULL ,c_0_9 BOOLEAN NOT NULL ) DUPLICATE KEY (c_0_0,c_0_1,c_0_2,c_0_3,c_0_4,c_0_5,c_0_6) PARTITION BY RANGE(c_0_1) (START ("2010-01-01") END ("2021-12-31") EVERY (INTERVAL 101 day)) DISTRIBUTED BY HASH (c_0_6) BUCKETS 3 properties("replication_num"="3") ;
CREATE TABLE t1 (c_1_0 DATE NOT NULL ) AGGREGATE KEY (c_1_0) PARTITION BY RANGE(c_1_0) (START ("2010-01-01") END ("2021-12-31") EVERY (INTERVAL 118 day)) DISTRIBUTED BY HASH (c_1_0) BUCKETS 3 properties("replication_num"="3") ;
CREATE TABLE IF NOT EXISTS t2 (c_2_0 DATETIME NOT NULL ,c_2_1 DATETIME NULL ,c_2_2 BOOLEAN REPLACE NOT NULL ) AGGREGATE KEY (c_2_0,c_2_1) PARTITION BY RANGE(c_2_0) (START ("2010-01-01") END ("2021-12-31") EVERY (INTERVAL 156 day)) DISTRIBUTED BY HASH (c_2_1,c_2_0) BUCKETS 3 properties("replication_num"="3") ;
DELETE FROM t0 WHERE c_0_2=45792817189563095.364029 AND c_0_5='' AND c_0_1='1970-01-03' AND c_0_4=false AND c_0_3=false AND c_0_6 IS NOT NULL;
INSERT INTO t0 VALUES (true, NULL, 40469810720448152.565537, false, true, 'HHj', 5, ["WRlWeTarJ", "E", "O3couJAJ0", "JitP", "", "Nu3jM", "", "b", "g1YQb", "v5AqU", "cd97BBuhw", "XC", "4"], '2009-12-31 08:00:14', false), (false, '2010-01-04', 32605644289607488.819378, false, true, 'Y', 0, ["", "", "", "gDUDen0", "gE", "", "", "0W26A6", ""], '2010-01-24 08:00:14', true);
INSERT INTO t1 VALUES ('2010-01-24'), ('2010-01-24'), ('2010-01-08');
CREATE VIEW v0 AS SELECT t1_15.c_1_0 FROM t1 AS t1_15;
CREATE VIEW v1 AS SELECT t1_15.c_1_0 FROM t1 AS t1_15 WHERE true;
CREATE VIEW v2 AS SELECT t1_15.c_1_0 FROM t1 AS t1_15 WHERE (NOT (true));
CREATE VIEW v3 AS SELECT t1_15.c_1_0 FROM t1 AS t1_15 WHERE ((true) IS NULL);
CREATE VIEW v4 AS SELECT t1_15.c_1_0 FROM t1 AS t1_15 WHERE true UNION ALL SELECT t1_15.c_1_0 FROM t1 AS t1_15 WHERE (NOT (true)) UNION ALL SELECT t1_15.c_1_0 FROM t1 AS t1_15 WHERE ((true) IS NULL);
CREATE VIEW v5 AS WITH with_t_0 as (SELECT t1_40.c_1_0 FROM t1 AS t1_40 WHERE NULL) , with_t_1 as (SELECT v2_44.c_1_0 FROM v2 AS v2_44 WHERE false) SELECT MICROSECONDS_ADD('1969-12-26', -248266219)  FROM v2 AS v2_44, v4 AS v4_46, t2 AS t2_41, with_t_0 AS with_t_0, with_t_1 AS with_t_1,  (SELECT v1_43.c_1_0 FROM v1 AS v1_43) subv1 LEFT ANTI JOIN (SELECT v0_42.c_1_0 FROM v0 AS v0_42 WHERE false) subv0 ON subv1.c_1_0 = subv0.c_1_0 AND subv1.c_1_0 = subv0.c_1_0 AND subv1.c_1_0 < subv0.c_1_0 AND subv1.c_1_0 < subv0.c_1_0 AND subv1.c_1_0 >= subv0.c_1_0 AND subv1.c_1_0 < CAST(0.9838377867844983 AS DATE )  AND subv1.c_1_0 != subv0.c_1_0 AND NULL != subv0.c_1_0,  (SELECT t0_39.c_0_0, t0_39.c_0_1, t0_39.c_0_2, t0_39.c_0_3, t0_39.c_0_4, t0_39.c_0_5, t0_39.c_0_6, t0_39.c_0_8, t0_39.c_0_9 FROM t0 AS t0_39 WHERE true) subt0 LEFT ANTI JOIN t1 AS t1_40 ON subt0.c_0_8 = t1_40.c_1_0 AND subt0.c_0_8 > t1_40.c_1_0 AND subt0.c_0_5 = t1_40.c_1_0 AND subt0.c_0_5 <= t1_40.c_1_0 AND subt0.c_0_8 <= t1_40.c_1_0 AND subt0.c_0_8 != MINUTES_SUB('1969-12-16', -497450797) ;
CREATE VIEW v5 AS SELECT v3_53.c_1_0 FROM v3 AS v3_53;
CREATE VIEW v6 AS SELECT v3_53.c_1_0 FROM v3 AS v3_53 WHERE CAST(v3_53.c_1_0 AS BOOLEAN ) ;
CREATE VIEW v7 AS SELECT v3_53.c_1_0 FROM v3 AS v3_53 WHERE (NOT (CAST(v3_53.c_1_0 AS BOOLEAN ) ));
CREATE VIEW v8 AS SELECT v3_53.c_1_0 FROM v3 AS v3_53 WHERE ((CAST(v3_53.c_1_0 AS BOOLEAN ) ) IS NULL);
CREATE VIEW v9 AS SELECT v3_53.c_1_0 FROM v3 AS v3_53 WHERE CAST(v3_53.c_1_0 AS BOOLEAN )  UNION ALL SELECT v3_53.c_1_0 FROM v3 AS v3_53 WHERE (NOT (CAST(v3_53.c_1_0 AS BOOLEAN ) )) UNION ALL SELECT v3_53.c_1_0 FROM v3 AS v3_53 WHERE ((CAST(v3_53.c_1_0 AS BOOLEAN ) ) IS NULL);
CREATE VIEW v63 AS SELECT subv5.c_1_0 FROM v7 AS v7_107, v0 AS v0_100,  (SELECT v5_105.c_1_0 FROM v5 AS v5_105) subv5 LEFT ANTI JOIN t1 AS t1_98 ON subv5.c_1_0 = t1_98.c_1_0 AND subv5.c_1_0 >= t1_98.c_1_0 AND subv5.c_1_0 != t1_98.c_1_0 AND '1969-12-19' >= t1_98.c_1_0,  v4 AS v4_104 LEFT ANTI JOIN v3 AS v3_103 ON v4_104.c_1_0 = v3_103.c_1_0 AND v4_104.c_1_0 > v3_103.c_1_0 AND v4_104.c_1_0 = v3_103.c_1_0 AND '1969-12-30' != DAYS_ADD('1969-12-30', -94607474) ;
CREATE VIEW v10 AS SELECT subv5.c_1_0 FROM v7 AS v7_107, v0 AS v0_100,  (SELECT v5_105.c_1_0 FROM v5 AS v5_105) subv5 LEFT ANTI JOIN t1 AS t1_98 ON subv5.c_1_0 = t1_98.c_1_0 AND subv5.c_1_0 >= t1_98.c_1_0 AND subv5.c_1_0 != t1_98.c_1_0 AND '1969-12-19' >= t1_98.c_1_0,  v4 AS v4_104 LEFT ANTI JOIN v3 AS v3_103 ON v4_104.c_1_0 = v3_103.c_1_0 AND v4_104.c_1_0 > v3_103.c_1_0 AND v4_104.c_1_0 = v3_103.c_1_0 AND '1969-12-30' != DAYS_ADD('1969-12-30', -94607474)  WHERE ((subv5.c_1_0) IS NOT NULL);
CREATE VIEW v11 AS SELECT subv5.c_1_0 FROM v7 AS v7_107, v0 AS v0_100,  (SELECT v5_105.c_1_0 FROM v5 AS v5_105) subv5 LEFT ANTI JOIN t1 AS t1_98 ON subv5.c_1_0 = t1_98.c_1_0 AND subv5.c_1_0 >= t1_98.c_1_0 AND subv5.c_1_0 != t1_98.c_1_0 AND '1969-12-19' >= t1_98.c_1_0,  v4 AS v4_104 LEFT ANTI JOIN v3 AS v3_103 ON v4_104.c_1_0 = v3_103.c_1_0 AND v4_104.c_1_0 > v3_103.c_1_0 AND v4_104.c_1_0 = v3_103.c_1_0 AND '1969-12-30' != DAYS_ADD('1969-12-30', -94607474)  WHERE (NOT (((subv5.c_1_0) IS NOT NULL)));
CREATE VIEW v35 AS SELECT subv5.c_1_0 FROM v7 AS v7_107, v0 AS v0_100,  (SELECT v5_105.c_1_0 FROM v5 AS v5_105) subv5 LEFT ANTI JOIN t1 AS t1_98 ON subv5.c_1_0 = t1_98.c_1_0 AND subv5.c_1_0 >= t1_98.c_1_0 AND subv5.c_1_0 != t1_98.c_1_0 AND '1969-12-19' >= t1_98.c_1_0,  v4 AS v4_104 LEFT ANTI JOIN v3 AS v3_103 ON v4_104.c_1_0 = v3_103.c_1_0 AND v4_104.c_1_0 > v3_103.c_1_0 AND v4_104.c_1_0 = v3_103.c_1_0 AND '1969-12-30' != DAYS_ADD('1969-12-30', -94607474)  WHERE ((((subv5.c_1_0) IS NOT NULL)) IS NULL);
CREATE VIEW v12 AS SELECT subv5.c_1_0 FROM v7 AS v7_107, v0 AS v0_100,  (SELECT v5_105.c_1_0 FROM v5 AS v5_105) subv5 LEFT ANTI JOIN t1 AS t1_98 ON subv5.c_1_0 = t1_98.c_1_0 AND subv5.c_1_0 >= t1_98.c_1_0 AND subv5.c_1_0 != t1_98.c_1_0 AND '1969-12-19' >= t1_98.c_1_0,  v4 AS v4_104 LEFT ANTI JOIN v3 AS v3_103 ON v4_104.c_1_0 = v3_103.c_1_0 AND v4_104.c_1_0 > v3_103.c_1_0 AND v4_104.c_1_0 = v3_103.c_1_0 AND '1969-12-30' != DAYS_ADD('1969-12-30', -94607474)  WHERE ((subv5.c_1_0) IS NOT NULL) UNION ALL SELECT subv5.c_1_0 FROM v7 AS v7_107, v0 AS v0_100,  (SELECT v5_105.c_1_0 FROM v5 AS v5_105) subv5 LEFT ANTI JOIN t1 AS t1_98 ON subv5.c_1_0 = t1_98.c_1_0 AND subv5.c_1_0 >= t1_98.c_1_0 AND subv5.c_1_0 != t1_98.c_1_0 AND '1969-12-19' >= t1_98.c_1_0,  v4 AS v4_104 LEFT ANTI JOIN v3 AS v3_103 ON v4_104.c_1_0 = v3_103.c_1_0 AND v4_104.c_1_0 > v3_103.c_1_0 AND v4_104.c_1_0 = v3_103.c_1_0 AND '1969-12-30' != DAYS_ADD('1969-12-30', -94607474)  WHERE (NOT (((subv5.c_1_0) IS NOT NULL))) UNION ALL SELECT subv5.c_1_0 FROM v7 AS v7_107, v0 AS v0_100,  (SELECT v5_105.c_1_0 FROM v5 AS v5_105) subv5 LEFT ANTI JOIN t1 AS t1_98 ON subv5.c_1_0 = t1_98.c_1_0 AND subv5.c_1_0 >= t1_98.c_1_0 AND subv5.c_1_0 != t1_98.c_1_0 AND '1969-12-19' >= t1_98.c_1_0,  v4 AS v4_104 LEFT ANTI JOIN v3 AS v3_103 ON v4_104.c_1_0 = v3_103.c_1_0 AND v4_104.c_1_0 > v3_103.c_1_0 AND v4_104.c_1_0 = v3_103.c_1_0 AND '1969-12-30' != DAYS_ADD('1969-12-30', -94607474)  WHERE ((((subv5.c_1_0) IS NOT NULL)) IS NULL);
CREATE VIEW v13 AS SELECT v8_188.c_1_0 FROM v8 AS v8_188, v10 AS v10_177, v3 AS v3_181, v7 AS v7_187, v0 AS v0_175, t0 AS t0_172, t2 AS t2_174;
CREATE VIEW v14 AS SELECT v8_188.c_1_0 FROM v8 AS v8_188, v10 AS v10_177, v3 AS v3_181, v7 AS v7_187, v0 AS v0_175, t0 AS t0_172, t2 AS t2_174 WHERE false;
CREATE VIEW v15 AS SELECT v8_188.c_1_0 FROM v8 AS v8_188, v10 AS v10_177, v3 AS v3_181, v7 AS v7_187, v0 AS v0_175, t0 AS t0_172, t2 AS t2_174 WHERE (NOT (false));
CREATE VIEW v16 AS SELECT v8_188.c_1_0 FROM v8 AS v8_188, v10 AS v10_177, v3 AS v3_181, v7 AS v7_187, v0 AS v0_175, t0 AS t0_172, t2 AS t2_174 WHERE ((false) IS NULL);
CREATE VIEW v17 AS SELECT v8_188.c_1_0 FROM v8 AS v8_188, v10 AS v10_177, v3 AS v3_181, v7 AS v7_187, v0 AS v0_175, t0 AS t0_172, t2 AS t2_174 WHERE false UNION ALL SELECT v8_188.c_1_0 FROM v8 AS v8_188, v10 AS v10_177, v3 AS v3_181, v7 AS v7_187, v0 AS v0_175, t0 AS t0_172, t2 AS t2_174 WHERE (NOT (false)) UNION ALL SELECT v8_188.c_1_0 FROM v8 AS v8_188, v10 AS v10_177, v3 AS v3_181, v7 AS v7_187, v0 AS v0_175, t0 AS t0_172, t2 AS t2_174 WHERE ((false) IS NULL);
CREATE VIEW v18 AS SELECT v11_278.c_1_0 FROM v11 AS v11_278, v13 AS v13_280, v17 AS v17_284, v9 AS v9_294, v15 AS v15_282, v10 AS v10_277, v2 AS v2_285, v5 AS v5_289, v8 AS v8_293, v63 AS v63_291, t1 AS t1_273, v1 AS v1_276, v12 AS v12_279, v14 AS v14_281, v4 AS v4_288, v3 AS v3_286, v35 AS v35_287, t0 AS t0_272, v7 AS v7_292;
CREATE VIEW v59 AS SELECT v11_278.c_1_0 FROM v11 AS v11_278, v13 AS v13_280, v17 AS v17_284, v9 AS v9_294, v15 AS v15_282, v10 AS v10_277, v2 AS v2_285, v5 AS v5_289, v8 AS v8_293, v63 AS v63_291, t1 AS t1_273, v1 AS v1_276, v12 AS v12_279, v14 AS v14_281, v4 AS v4_288, v3 AS v3_286, v35 AS v35_287, t0 AS t0_272, v7 AS v7_292 WHERE CASE  WHEN CAST('1970-01-14' AS BOOLEAN )  THEN (((v13_280.c_1_0) IS NOT NULL)) IN (false, false)  ELSE true END;

```
```
fe.log:
com.starrocks.common.DdlException: failed to init view stmt
        at com.starrocks.server.LocalMetastore.createView(LocalMetastore.java:3377) ~[starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr.createView(GlobalStateMgr.java:2767) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.DdlExecutor.execute(DdlExecutor.java:201) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.StmtExecutor.handleDdlStmt(StmtExecutor.java:924) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.StmtExecutor.execute(StmtExecutor.java:438) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.handleQuery(ConnectProcessor.java:288) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.dispatch(ConnectProcessor.java:436) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.processOnce(ConnectProcessor.java:672) ~[starrocks-fe.jar:?]
        at com.starrocks.mysql.nio.ReadListener.lambda$handleEvent$0(ReadListener.java:55) ~[starrocks-fe.jar:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_252]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_252]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_252]
Caused by: com.starrocks.common.UserException: Failed to parse view-definition statement of view: v0
        at com.starrocks.catalog.View.init(View.java:203) ~[starrocks-fe.jar:?]
        at com.starrocks.server.LocalMetastore.createView(LocalMetastore.java:3375) ~[starrocks-fe.jar:?]
        ... 11 more
Caused by: com.starrocks.sql.parser.ParsingException: no viable alternative at input 'IN ((SELECT `t1_3356`.`c_1_0` AS `c_1_0` FROM `default_cluster:v19`.`t3` AS `t3_3358` , `default_cluster:v19`.`t2` AS `t2_3357`  , `default_cluster:v19`.`t1` AS `t1_3356` LEFT SEMI JOIN (SELECT `t0_3355`.`c_0_0` AS `c_0_0`, `t0_3355`.`c_0_1` AS `c_0_1`, `t0_3355`.`c_0_2` AS `c_0_2`, `t0_3355`.`c_0_3` AS `c_0_3`, `t0_3355`.`c_0_4` AS `c_0_4`, `t0_3355`.`c_0_5` AS `c_0_5`, `t0_3355`.`c_0_9` AS `c_0_9`, `t0_3355`.`c_0_10` AS `c_0_10` FROM `default_cluster:v19`.`t0` AS `t0_3355` WHERE FALSE) subt0 ON (`t1_3356`.`c_1_2` = `subt0`.`c_0_1`) AND ('1969-12-17' != `subt0`.`c_0_1`)  WHERE '1970-01-05' IN ('1970-01-03') IN'
        at com.starrocks.sql.parser.ErrorHandler.syntaxError(ErrorHandler.java:12) ~[starrocks-fe.jar:?]
        at org.antlr.v4.runtime.ProxyErrorListener.syntaxError(ProxyErrorListener.java:41) ~[antlr4-runtime-4.9.2.jar:4.9.2]
        at org.antlr.v4.runtime.Parser.notifyErrorListeners(Parser.java:544) ~[antlr4-runtime-4.9.2.jar:4.9.2]
        at org.antlr.v4.runtime.DefaultErrorStrategy.reportNoViableAlternative(DefaultErrorStrategy.java:310) ~[antlr4-runtime-4.9.2.jar:4.9.2]
        at org.antlr.v4.runtime.DefaultErrorStrategy.reportError(DefaultErrorStrategy.java:136) ~[antlr4-runtime-4.9.2.jar:4.9.2]
        at com.starrocks.sql.parser.StarRocksParser.predicateOperations(StarRocksParser.java:10104) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.parser.StarRocksParser.predicate(StarRocksParser.java:9847) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.parser.StarRocksParser.booleanExpression(StarRocksParser.java:9721) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.parser.StarRocksParser.expression(StarRocksParser.java:9501) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.parser.StarRocksParser.querySpecification(StarRocksParser.java:6588) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.parser.StarRocksParser.queryPrimary(StarRocksParser.java:6151) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.parser.StarRocksParser.queryTerm(StarRocksParser.java:6000) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.parser.StarRocksParser.queryNoWith(StarRocksParser.java:5861) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.parser.StarRocksParser.queryBody(StarRocksParser.java:5738) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.parser.StarRocksParser.commonTableExpression(StarRocksParser.java:7157) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.parser.StarRocksParser.withClause(StarRocksParser.java:5799) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.parser.StarRocksParser.queryBody(StarRocksParser.java:5733) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.parser.StarRocksParser.queryStatement(StarRocksParser.java:5670) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.parser.StarRocksParser.statement(StarRocksParser.java:1312) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.parser.StarRocksParser.singleStatement(StarRocksParser.java:455) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.parser.StarRocksParser.sqlStatements(StarRocksParser.java:377) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.parser.SqlParser.parse(SqlParser.java:32) ~[starrocks-fe.jar:?]
        at com.starrocks.catalog.View.init(View.java:195) ~[starrocks-fe.jar:?]
        at com.starrocks.server.LocalMetastore.createView(LocalMetastore.java:3375) ~[starrocks-fe.jar:?]
```



### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
 - mysql> select current_version();
+----------------------+
| current_version()    |
+----------------------+
| MAIN-RELEASE d12a28a |
+----------------------+
- cluster: sqlancer_01_release

--- a/fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AST2SQL.java
+++ b/fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AST2SQL.java
@@ -391,14 +391,14 @@ public class AST2SQL {
             StringBuilder output = new StringBuilder("CASE");
             int childIdx = 0;
             if (hasCaseExpr) {
-                output.append(" ").append(visit(node.getChild(childIdx++)));
+                output.append(" ").append(printWithParentheses(node.getChild(childIdx++)));
             }
             while (childIdx + 2 <= node.getChildren().size()) {
-                output.append(" WHEN ").append(visit(node.getChild(childIdx++)));
-                output.append(" THEN ").append(visit(node.getChild(childIdx++)));
+                output.append(" WHEN ").append(printWithParentheses(node.getChild(childIdx++)));
+                output.append(" THEN ").append(printWithParentheses(node.getChild(childIdx++)));
             }
             if (hasElseExpr) {
-                output.append(" ELSE ").append(visit(node.getChild(node.getChildren().size() - 1)));
+                output.append(" ELSE ").append(printWithParentheses(node.getChild(node.getChildren().size() - 1)));
             }
             output.append(" END");
             return output.toString();
@@ -410,15 +410,14 @@ public class AST2SQL {
             if (isImplicit) {
                 return visit(node.getChild(0));
             }
-            return "CAST(" + visit(node.getChild(0)) + " AS " + node.getTargetTypeDef().toString() + ")";
+            return "CAST(" + printWithParentheses(node.getChild(0)) + " AS " + node.getTargetTypeDef().toString() + ")";
         }
 
         public String visitCompoundPredicate(CompoundPredicate node, Void context) {
             StringBuilder sqlBuilder = new StringBuilder();
             if (CompoundPredicate.Operator.NOT.equals(node.getOp())) {
-                sqlBuilder.append("( NOT ");
+                sqlBuilder.append("NOT ");
                 sqlBuilder.append(printWithParentheses(node.getChild(0)));
-                sqlBuilder.append(" )");
             } else {
                 sqlBuilder.append(printWithParentheses(node.getChild(0)));
                 sqlBuilder.append(" ").append(node.getOp()).append(" ");
@@ -439,7 +438,7 @@ public class AST2SQL {
 
             }
             strBuilder.append("EXISTS ");
-            strBuilder.append(visit(node.getChild(0)));
+            strBuilder.append(printWithParentheses(node.getChild(0)));
             return strBuilder.toString();
         }
 
@@ -477,9 +476,9 @@ public class AST2SQL {
         public String visitInPredicate(InPredicate node, Void context) {
             StringBuilder strBuilder = new StringBuilder();
             String notStr = (node.isNotIn()) ? "NOT " : "";
-            strBuilder.append(visit(node.getChild(0))).append(" ").append(notStr).append("IN (");
+            strBuilder.append(printWithParentheses(node.getChild(0))).append(" ").append(notStr).append("IN (");
             for (int i = 1; i < node.getChildren().size(); ++i) {
-                strBuilder.append(visit(node.getChild(i)));
+                strBuilder.append(printWithParentheses(node.getChild(i)));
                 strBuilder.append((i + 1 != node.getChildren().size()) ? ", " : "");
             }
             strBuilder.append(")");
@@ -487,11 +486,12 @@ public class AST2SQL {
         }
 
         public String visitIsNullPredicate(IsNullPredicate node, Void context) {
-            return visit(node.getChild(0)) + (node.isNotNull() ? " IS NOT NULL" : " IS NULL");
+            return printWithParentheses(node.getChild(0)) + (node.isNotNull() ? " IS NOT NULL" : " IS NULL");
         }
 
         public String visitLikePredicate(LikePredicate node, Void context) {
-            return visit(node.getChild(0)) + " " + node.getOp() + " " + visit(node.getChild(1));
+            return printWithParentheses(node.getChild(0))
+                    + " " + node.getOp() + " " + printWithParentheses(node.getChild(1));
         }
 
         public String visitLiteral(LiteralExpr node, Void context) {

--- a/fe/fe-core/src/test/java/com/starrocks/sql/analyzer/AST2SQLTest.java
+++ b/fe/fe-core/src/test/java/com/starrocks/sql/analyzer/AST2SQLTest.java
@@ -32,7 +32,7 @@ public class AST2SQLTest {
             Assert.assertTrue(baseStmt instanceof CreateViewStmt);
             CreateViewStmt viewStmt = (CreateViewStmt) baseStmt;
             Assert.assertEquals(viewStmt.getInlineViewDef(),
-                    "SELECT `default_cluster:test`.`t0`.`v1` AS `v1` FROM `default_cluster:test`.`t0` WHERE ( NOT FALSE ) IS NULL");
+                    "SELECT `default_cluster:test`.`t0`.`v1` AS `v1` FROM `default_cluster:test`.`t0` WHERE (NOT FALSE) IS NULL");
         }
         {
             String sql;
@@ -47,7 +47,7 @@ public class AST2SQLTest {
             Assert.assertTrue(baseStmt instanceof CreateViewStmt);
             CreateViewStmt viewStmt = (CreateViewStmt) baseStmt;
             Assert.assertEquals(viewStmt.getInlineViewDef(),
-                    "SELECT `default_cluster:test`.`t0`.`v1` AS `v1` FROM `default_cluster:test`.`t0` WHERE ( NOT FALSE ) IS NOT NULL");
+                    "SELECT `default_cluster:test`.`t0`.`v1` AS `v1` FROM `default_cluster:test`.`t0` WHERE (NOT FALSE) IS NOT NULL");
         }
     }
 }

--- a/fe/fe-core/src/test/java/com/starrocks/sql/analyzer/AnalyzeSingleTest.java
+++ b/fe/fe-core/src/test/java/com/starrocks/sql/analyzer/AnalyzeSingleTest.java
@@ -444,7 +444,7 @@ public class AnalyzeSingleTest {
                 connectContext.getSessionVariable().getSqlMode()).get(0);
         Analyzer.analyze(statementBase, connectContext);
         Assert.assertEquals(
-                "SELECT * FROM default_cluster:test.tall WHERE ta LIKE concat(concat('h', 'a', 'i'), '%')",
+                "SELECT * FROM default_cluster:test.tall WHERE ta LIKE (concat(concat('h', 'a', 'i'), '%'))",
                 AST2SQL.toString(statementBase));
 
         connectContext.getSessionVariable().setSqlMode(0);
@@ -452,11 +452,11 @@ public class AnalyzeSingleTest {
                 connectContext.getSessionVariable().getSqlMode()).get(0);
         Analyzer.analyze(statementBase, connectContext);
         Assert.assertEquals(
-                "SELECT * FROM default_cluster:test.tall WHERE (ta LIKE concat('h', 'a', 'i')) OR TRUE",
+                "SELECT * FROM default_cluster:test.tall WHERE (ta LIKE (concat('h', 'a', 'i'))) OR TRUE",
                 AST2SQL.toString(statementBase));
 
         analyzeFail("select * from  tall where ta like concat(\"h\", \"a\", \"i\")||'%'",
-                "LIKE concat('h', 'a', 'i')) OR '%'' part of predicate ''%'' should return type 'BOOLEAN'");
+                "LIKE (concat('h', 'a', 'i'))) OR '%'' part of predicate ''%'' should return type 'BOOLEAN'");
 
         connectContext.getSessionVariable().setSqlMode(SqlModeHelper.MODE_SORT_NULLS_LAST);
         statementBase = SqlParser.parse("select * from  tall order by ta",

--- a/fe/fe-core/src/test/java/com/starrocks/sql/plan/ViewPlanTest.java
+++ b/fe/fe-core/src/test/java/com/starrocks/sql/plan/ViewPlanTest.java
@@ -1556,6 +1556,12 @@ public class ViewPlanTest extends PlanTestBase {
     }
 
     @Test
+    public void test315() throws Exception {
+        String sql = "select * from t0 where case when true then (v1 is null ) in (true,false) else true end";
+        testView(sql);
+    }
+
+    @Test
     public void testAliasView() throws Exception {
         String sql = "select * from (select a.A, count(*) as cnt from ( SELECT 1 AS A UNION ALL SELECT 2 UNION ALL " +
                 "SELECT 2 UNION ALL SELECT 5 UNION ALL SELECT 3 UNION ALL SELECT 0 AS A ) a group by a.A order by a.A ) b;";

